lib/nostrum/shard/event.ex

```diff
@@ -45,11 +42,11 @@ defmodule Nostrum.Shard.Event do
     else
       Logger.debug "IDENTIFYING"
       identify(self())
-
-      heartbeat(self(), payload.d.heartbeat_interval)
     end
 
-    state
+    heartbeat_task = Task.async(fn -> heartbeat(self(), payload.d.heartbeat_interval) end)
```

Comment:
The call to `self()` here will be in the context of the task, when it should be in the context of the shard process.